### PR TITLE
feat(stackable-versioned): Adjust generated module and container names

### PR DESCRIPTION
**Pull request by Techassi**
# Description

Tracked by https://github.com/stackabletech/issues/issues/507, follow-up of #764 and #793

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

### CODE REVIEW

**Comment by Techassi** [comment]
Seems like we currently cannot use the `heck` crate for case conversions: https://github.com/withoutboats/heck/issues/58.

**Comment by Techassi** [comment]
The generated code now looks like this:

```rust
#[versioned(
    version(name = "v1alpha1"),
    version(name = "v1beta1"),
    version(name = "v1"),
    version(name = "v2"),
    version(name = "v3")
)]
struct Foo {
    /// My docs
    #[versioned(
        added(since = "v1alpha1"),
        renamed(since = "v1beta1", from = "jjj"),
        deprecated(since = "v2", note = "not empty")
    )]
    deprecated_bar: usize,
    baz: bool,
}

// Produces ...

#[automatically_derived]
pub mod foo {
    pub struct V1Alpha1 {
        pub jjj: usize,
        pub baz: bool,
    }
    pub struct V1Beta1 {
        pub bar: usize,
        pub baz: bool,
    }
    pub struct V1 {
        pub bar: usize,
        pub baz: bool,
    }
    pub struct V2 {
        #[deprecated = "not empty"]
        pub deprecated_bar: usize,
        pub baz: bool,
    }
    pub struct V3 {
        // Missing deprecated attribute will be fixed
        pub deprecated_bar: usize,
        pub baz: bool,
    }
}
pub type Foo = foo::V3;
```

File: crates/stackable-versioned/CHANGELOG.md [hunk 1]
```diff
@@ -4,11 +4,13 @@ All notable changes to this project will be documented in this file.
 
 ## [Unreleased]
 
+- Adjust generated module and container names ([#CHANGEME]).
```
Comment (NickLarsenNZ) [inline crates/stackable-versioned/CHANGELOG.md:7]:
```suggestion
- Adjust generated module and container names ([#797]).
```

File: crates/stackable-versioned/CHANGELOG.md [hunk 2]
```diff
@@ -4,11 +4,13 @@ All notable changes to this project will be documented in this file.
...
 - Change from derive macro to attribute macro to be able to generate code
-  _in place_ instead of _appending_ new code ([#CHANGEME]).
+  _in place_ instead of _appending_ new code ([#793]).
 - Improve action chain generation ([#784]).
 
 [#784](https://github.com/stackabletech/operator-rs/pull/784)
+[#793](https://github.com/stackabletech/operator-rs/pull/793)
 [#CHANGEME](https://github.com/stackabletech/operator-rs/pull/CHANGEME)
```
Comment (NickLarsenNZ) [inline crates/stackable-versioned/CHANGELOG.md:14]:
```suggestion
[#797](https://github.com/stackabletech/operator-rs/pull/797)
```

**Review by NickLarsenNZ — APPROVED** [review]
LGTM

File: crates/stackable-versioned/CHANGELOG.md (hunk 1 above)
Comment (Techassi) [reply crates/stackable-versioned/CHANGELOG.md:7]: Ah yeah, totally forgot that. Thx!

**Comment by sbernauer** [comment]
I guess we should also take https://github.com/stackabletech/operator-rs/pull/793#issuecomment-2129232150 into consideration

**Comment by Techassi** [comment]
Putting this PR on hold for now, as some more discussion is happening over at #793.

**Comment by Techassi** [comment]
A decision in #793 was reached. We will stick with the current naming scheme and thus this PR will be closed.

**Pull request closed without merging**